ArmPkg/Drivers/SmmuDxe/IoMmu.c

```diff
@@ -51,38 +52,30 @@ STATIC
 EFI_STATUS
 UpdateFlags (
   IN PAGE_TABLE  *Table,
-  IN BOOLEAN     SetReadWriteFlagsOnly,
   IN UINT16      Flags,
   IN UINT32      Index
   )
 {
-  EFI_STATUS  Status;
+  UINT64  Entry;
 
   if ((Table == NULL) || ((Flags & ~PAGE_TABLE_BLOCK_OFFSET) != 0) || (Index >= PAGE_TABLE_SIZE)) {
     DEBUG ((DEBUG_ERROR, "%a: Invalid parameter.\n", __func__));
-    Status = EFI_INVALID_PARAMETER;
-    ASSERT_EFI_ERROR (Status);
-    return Status;
+    ASSERT_EFI_ERROR (EFI_INVALID_PARAMETER);
+    return EFI_INVALID_PARAMETER;
   }
 
-  Status = EFI_SUCCESS;
-
-  // This boolean is used to explicity update the R/W bits in the page table entry.
   // Allows clearing the R/W bits without affecting the other bits in the entry.
-  if (SetReadWriteFlagsOnly) {
-    if (Flags != 0) {
-      // Set R/W bits in page table entry
-      Table->Entries[Index] |= Flags;
-    } else {
-      // Clear R/W bits in page table entry
-      Table->Entries[Index] &= ~(PAGE_TABLE_READ_BIT | PAGE_TABLE_WRITE_BIT);
-    }
-  } else {
+  if (Flags != 0) {
     // Set R/W bits in page table entry
-    Table->Entries[Index] |= Flags;
+    Entry = Table->Entries[Index] | Flags;

```

Comment:
Should this not ensure that they are cleared first. i.e. if the caller provides 0b10 but its currently set to 0b01 is the expectation that is should then be 0b11 or 0b10?